src/main/webapp/script.js

```diff
@@ -196,11 +210,23 @@ function displayMatches() {
 function createCardElement(userID) {
   const cardDiv = document.createElement("div");
   fetch('/user-data?id=' + userID).then(response => response.json()).then((userinfo) => {
+    console.log(userinfo);
     if (userinfo === null) {
         return;
     }
     cardDiv.className = "col card m-5 card-container";
-    const profileImage = createImgElement("images/noBlobStoreImage2.jpg");
+    let profileImage;
+    let foundImage = false;
+    for (let i = 0; i < userinfo.blobkeys.length; i++) {
+      if (userinfo.blobkeys[i] != "") {
+        foundImage = true;
+        profileImage = createImageFromBlobstore(userinfo.blobkeys[i]);
+        break;
+      }
+    }
```

Comment:
I'd move this logic to its own function `getProfileImage` that returns either an image element, or a nullable blobKey.